tests/pytest/test_capture.py

```diff
@@ -88,26 +88,42 @@ def test_output_emitting_unicode(testdir, monkeypatch):
 TRACEBACKHIDE_TIMEOUT = """
 from hypothesis import given, settings
 from hypothesis.strategies import integers
+from hypothesis.errors import HypothesisDeprecationWarning
+
 import time
+import warnings
+import pytest
 
-@given(integers())
-@settings(timeout=1)
-def test_timeout_traceback_is_hidden(i):
-    time.sleep(1.1)
+
+def test_timeout_traceback_is_hidden():
+    with warnings.catch_warnings(record=True):
+        warnings.simplefilter('ignore', HypothesisDeprecationWarning)
+        @given(integers())
+        @settings(timeout=1)
+        def inner(i):
+            time.sleep(1.1)
+        inner()
 """
 
 
-def get_line_num(token, result):
+def get_line_num(token, result, skip_n=0):
     for i, line in enumerate(result.stdout.lines):
         if token in line:
-            return i
-    assert False, 'Token %r not found' % token
+            if skip_n == 0:
+                return i
+            else:
+                skip_n -= 1
+    assert False, \
+        'Token %r not found (after skipping %r appearances)' % (token, skip_n)
 
 
 def test_timeout_traceback_is_hidden(testdir):
     script = testdir.makepyfile(TRACEBACKHIDE_TIMEOUT)
     result = testdir.runpytest(script, '--verbose')
-    def_line = get_line_num('def test_timeout_traceback_is_hidden', result)
+    # `def inner` shows up in the output twice: once when pytest shows us the
```

Comment:
@Zac-HD This is a result of following your suggestion from https://github.com/HypothesisWorks/hypothesis-python/pull/1169#discussion_r174993238: With your version, it turns out `def inner` appears twice.

I still like your version better (entirely avoids the need for my confusing fixture!), but the skip business is a bit annoying. I'm definitely open to better ideas if you have any!

Reply:
My "better idea" is just to put up with this for a few more weeks (or months), and tear out everything related to the timeout setting in Hypothesis 4.0 - it is after all deprecated!

(lots of satisfying things like that in #1172, even if I haven't gotten to this one yet)